tests/unit/api/BackbeatAPI.spec.js

```diff
@@ -210,4 +220,40 @@ describe('BackbeatAPI', () => {
             assert.deepStrictEqual(sites, params.expected);
         });
     });
+
+    describe('_setZookeeper', () => {
+        let zkManagerArgs;
+        class MockZookeeperManager {
+            constructor(url, options, logger) {
+                zkManagerArgs = { url, options, logger };
+                this.once = sinon.stub().callsFake((event, callback) => {
+                    if (event === 'connected') {
+                        callback();
+                    }
+                });
+                this.removeAllListeners = sinon.stub();
+            }
+        }
+
+        after(() => {
+            sinon.restore();
+        });
+
+        it('should use connectionString directly if this._queuePopulator.mongo exists', done => {
+            bbapi._setZookeeper(() => {
+                assert(zkManagerArgs);
+                assert.strictEqual(zkManagerArgs.url, '127.0.0.1:2181');
+                done();
+            }, MockZookeeperManager);
+        });
+
+        it('should append zookeeperPath to connectionString if this._queuePopulator.mongo does not exist', done => {
+            delete bbapi._queuePopulator.mongo;
```

Comment:
As we should not rely on the order ot unit tests for them to pass, I think this should be reverted at the end of the test (or just mock the value)